tests/tests_main.cpp

```diff
@@ -0,0 +1,7 @@
+// Following the recommendations from catch2 we have a single main file that
+// only defines the main. This should help in keeping (re)compilation times
+// somewhat lower than if we define it directly in the test file
+// See also: https://github.com/catchorg/Catch2/blob/devel/docs/slow-compiles.md
+
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
```

Comment:
```suggestion
#define CATCH_CONFIG_FAST_COMPILE 
#include <catch2/catch_test_macros.hpp>
```
For Catch2, I'd  suggest to already work with the v3.x series (only pre-releases yet, and actually only the master works for me). The reason being is that it ships a static library that will help us avoid recompiling Catch2 for every package. See https://github.com/key4hep/k4SimDelphes/pull/17